index.js

```diff
@@ -106,6 +107,28 @@ async function fastifyView (fastify, opts) {
 
   const renderer = renders[type] ? renders[type] : renders._default
 
+  async function asyncRender (page) {
+    if (!page) {
+      throw new Error('Missing page')
+    }
+
+    let result = await renderer.apply(this, arguments)
+
+    if (minify && !isPathExcludedMinification(this)) {
+      result = await minify(result, globalOptions.htmlMinifierOptions)
+    }
+
+    if (!this.getHeader('Content-Type')) {
+      this.header('Content-Type', 'text/html; charset=' + charset)
+    }
```

Comment:
Oh I get what you mean now, seems a little hacky but performant I guess :)

Can you please try this to see if it impacts performance? It could _maybe_ have an impact if we call this method multiple times

```suggestion
    this.header?.('Content-Type', 'text/html; charset=' + charset)
```

Reply:
Can't do that exactly because existing logic/tests ensure header is only set if not already set. Pushed something along these lines after confirming no regression is benchmarks.

Reply:
I see, thanks for the change, it is easier to understand for me